qiskit_experiments/framework/base_experiment.py

```diff
@@ -31,6 +32,55 @@
 from qiskit_experiments.framework.configs import ExperimentConfig
 
 
+def cached_method(method):
+    """Decorator to cache the return value of a BaseExperiment method.
+
+    This stores the output of a method in the experiment object instance
+    in a `_cache` dict attribute. Note that the value is cached only on
+    the object instance method name, not any values of its arguments.
+
+    The cache can be cleared by calling :meth:`.BaseExperiment.cache_clear`.
+    """
+
+    @wraps(method)
+    def wrapped_method(self, *args, **kwargs):
+        name = f"{type(self).__name__}.{method.__name__}"
+
+        # making a tuple from the options value.
+        options_dict = vars(self.experiment_options)
+        cache_key = tuple(options_dict.values()) + tuple([name])
+        for key, val in options_dict.items():
+            if isinstance(val, list):  # pylint: disable=isinstance-second-argument-not-valid-type
+                val = tuple(val)
+                options_dict[key] = val
+                cache_key = tuple(options_dict.values()) + tuple([name])
+            if isinstance(  # pylint: disable=isinstance-second-argument-not-valid-type
+                val, Hashable
+            ):
+                continue
+            # if one of the values in option isn't hashable, we raise a warning and we use the name as
```

Comment:
Since you've already made the effort of walking over the options one-by-one, you can just remove the non-hashable options, while keeping the hashable ones.